thenewboston_node/blockchain/views/blockchain_states_meta.py

```diff
@@ -0,0 +1,68 @@
+from drf_spectacular.utils import OpenApiParameter, extend_schema
+from rest_framework.exceptions import NotFound, ParseError
+from rest_framework.response import Response
+from rest_framework.viewsets import ViewSet
+
+from thenewboston_node.blockchain import exceptions
+from thenewboston_node.blockchain.pagination import CustomLimitOffsetPagination
+from thenewboston_node.blockchain.serializers.blockchain_states_meta import BlockchainStatesMetaSerializer
+from thenewboston_node.business_logic.blockchain.base import BlockchainBase
+from thenewboston_node.business_logic.blockchain.file_blockchain import FileBlockchain
+
+SORT_DIRECTION_QUERY_MAP = {
+    'asc': 1,
+    'desc': -1,
+}
+
+
+class BlockchainStatesMetaViewSet(ViewSet):
+    serializer_class = BlockchainStatesMetaSerializer
+    pagination_class = CustomLimitOffsetPagination
+    lookup_value_regex = r'(-)?\d+'
+
+    @extend_schema(parameters=[OpenApiParameter('id', location=OpenApiParameter.PATH, description='Block number')])
+    def retrieve(self, request, pk: str = None):
+        assert pk is not None
+        block_number = int(pk)
```

Comment:
this may lead to HTTP500, but should be HTTP400 instead

Reply:
Can't we use https://www.django-rest-framework.org/api-guide/generic-views/#generic-views or our own alternatives?